### PR TITLE
Fix no-op pre prepare

### PR DESCRIPTION
**Pull request by olgavrou**
Resolves problem described in #839 

The problem here is that the no-op pre-prepare digests were not being calculated consistently during view changes (no-op pre-prepares are created per-replica vs created by one replica and broadcasted to other replicas as is done in normal execution) and so the no-op pre-prepares would never complete.

This fixes that and also puts the no-op pre-prepares into the ledger during view changes. In order for the no-op pp's to go into the ledger, we need them to not be signed, as each no-op pre-prepare will be signed by the replica that created it, leading eventually to an inconsistent merkle tree. 

These pre-prepares will be signed _eventually_ as part of the next pre-prepare.

### CODE REVIEW

File: src/consensus/pbft/libbyz/pre_prepare.cpp [hunk 1]
```diff
@@ -102,7 +103,10 @@ Pre_prepare::Pre_prepare(
 
   if (prepared_cert == nullptr)
   {
+    // if there are no pre-prepare proofs (e.g. no-op pre-prepare)
+    // then zero out the rest of the allocated message
     rep().num_prev_pp_sig = 0;
+    std::memset(proofs(), 0, size() - size_to_proofs());
```
Comment (ashamis) [inline src/consensus/pbft/libbyz/pre_prepare.cpp:109]: std::fill

Comment (ashamis) [inline src/consensus/pbft/libbyz/pre_prepare.cpp:109]: please use std::fill

**Comment by olgavrou** [comment]
@achamayou that is not a bad idea, I'll see how easy/hard it is to add that. In any case when the view change test is enabled this code path should be crossed in every view change.

**Comment by olgavrou** [comment]
Adding unit test in separate PR

**Pull request merged**